app/domain/authentication/authn_k8s/kubectl_exec.rb

```diff
@@ -140,12 +151,20 @@ def base_query_string_parts
       end
 
       def server_url(cmds, stdin)
-        api_uri = kubeclient.api_endpoint
+        api_uri = kubectl_client.api_endpoint
```

Comment:
I think it should be `kube_apiserver`, since it refers to this: https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver/

Reply:
this variable refers to a `Kubeclient::Client` object.

Reply:
kubeclient is not kubectl.
Although kubectl is one of the k8s clients, I think the k8s client referred to here is a ruby client; kubectl is a k8s client written in `go`.

Reply:
ah i see. thanks, i'll update it.

Reply:
You still have some `kubectl` around.
Like in: app/domain/authentication/authn_k8s/copy_text_to_file_in_container.rb
Maybe you could clean it up to make sure the code is readable and no one will confuse between `kubectl` and `kubeclient`.
What do you think?

Reply:
in this case it is `kubectl` as we perform `kubectl_exec`. am i missing something?

Reply:
@orenbm The name `kubectl` is misleading, as `kubectl` is a k8s client application written in go (the command-line tool).
The k8s client here is probably has nothing to do with `kubectl`.
So even `kubectl_exec` is not accurate.
Maybe this is going too deep for this PR, I opened an issue to handle it: #1855